app/code/community/Drip/Connect/Model/Observer/Customer/GuestSubscriberCreated.php

```diff
@@ -9,16 +9,24 @@ class Drip_Connect_Model_Observer_Customer_GuestSubscriberCreated extends Drip_C
      */
     protected function executeWhenEnabled($observer)
     {
-        if (! Mage::registry(self::REGISTRY_KEY_NEW_GUEST_SUBSCRIBER)) {
-            return;
-        }
-
         $email = Mage::app()->getRequest()->getParam('email');
+
         $subscriber = Mage::getModel('newsletter/subscriber')->loadByEmail($email);
 
         $config = Drip_Connect_Model_Configuration::forCurrentScope();
 
-        $this->proceedGuestSubscriberNew($subscriber, $config, $subscriber->isSubscribed());
+        $customer = Mage::helper('drip_connect')->getCustomerByEmail($email, $config);
+        if ($customer->getId() !== null) {
```

Comment:
I wonder whether it would improve clarity to switch the cases around and have the `if` statement be `$customer->getId() === null`…